specs/protocol/identifiers.md

```diff
@@ -25,7 +25,7 @@ Is represented as an _outpoint_: a pair of [transaction ID](#transaction-id) as
 
 ### Deposit ID
 
-The UTXO ID of a deposit is computed as the [hash](./cryptographic_primitives.md#hashing) of TODO.
+The UTXO ID of a deposit is computed as the [hash](./cryptographic_primitives.md#hashing) of the depoitor `address`, token `address`, token precision `u8` and deposit nonce `u256`.
```

Comment:
1. There's no notion of Solidity types in the FuelVM. Specifically, the specs only use `uint8`, `uint16`, uint32`, uint64`, `byte[32]`, and `byte[]`.
2. You also have to specify how these are serialized (you can point to the canonical serialization format): https://github.com/FuelLabs/fuel-specs/blob/84cff9a590a28d7c5e6cb62d91b52e8e30c6ef67/specs/protocol/tx_format.md#transaction.

Reply:
I've updated with references to serialization processes. Does it make sense to hash all of these values though? Really only the deposit nonce is enough to make it unique, but we'd need to add some sort of extra salt to avoid collision of common hashes (hash(0x0001), hash(0x0002), etc.) so I can see it as being ok. It's also useful as a tool to prove the params of a deposit given it's ID

Reply:
It does make sense so that the deposit is self-verifying. Using nonces alone to represent anything other than ordering is actually pretty dangerous due to re-orgs and frontrunning: if someone can manipulate the order of txs to make them different than you expect, then they can convince you of something invalid that you have no obvious way of knowing is invalid.

Reply:
As for domain separation...there's an issue for that! #201. You could leave doing that to a follow-up PR.

Reply:
Deposits will not be able to be used until they are finalized, I wouldn't recommend SDK to include it inside the next tx until we are sure that finalization is done. 
We can make that contract sends us nonce as id, while we mutate this into the id (hash of all data) that we want to be presented to users/sdk?